hazelcast/src/main/java/com/hazelcast/sql/impl/type/QueryDataType.java

```diff
@@ -0,0 +1,163 @@
+import com.hazelcast.sql.impl.type.converter.StringConverter;
+import com.hazelcast.sql.impl.type.converter.ZonedDateTimeConverter;
+
+import java.io.IOException;
+
+/**
+ * Data type represents a type of concrete expression which is based on some basic data type.
+ */
+public class QueryDataType implements IdentifiedDataSerializable {
+    public static final int PRECISION_BIT = 1;
+    public static final int PRECISION_TINYINT = 4;
+    public static final int PRECISION_SMALLINT = 7;
+    public static final int PRECISION_INT = 11;
+    public static final int PRECISION_BIGINT = 20;
+    public static final int PRECISION_UNLIMITED = -1;
+
+    public static final QueryDataType LATE = new QueryDataType(LateConverter.INSTANCE);
+
+    public static final QueryDataType VARCHAR = new QueryDataType(StringConverter.INSTANCE);
+    public static final QueryDataType VARCHAR_CHARACTER = new QueryDataType(CharacterConverter.INSTANCE);
+
+    public static final QueryDataType BIT = new QueryDataType(BooleanConverter.INSTANCE, PRECISION_BIT);
+    public static final QueryDataType TINYINT = new QueryDataType(ByteConverter.INSTANCE, PRECISION_TINYINT);
+    public static final QueryDataType SMALLINT = new QueryDataType(ShortConverter.INSTANCE, PRECISION_SMALLINT);
+    public static final QueryDataType INT = new QueryDataType(IntegerConverter.INSTANCE, PRECISION_INT);
+    public static final QueryDataType BIGINT = new QueryDataType(LongConverter.INSTANCE, PRECISION_BIGINT);
+    public static final QueryDataType DECIMAL = new QueryDataType(BigDecimalConverter.INSTANCE);
+    public static final QueryDataType DECIMAL_BIG_INTEGER = new QueryDataType(BigIntegerConverter.INSTANCE);
+    public static final QueryDataType REAL = new QueryDataType(FloatConverter.INSTANCE);
+    public static final QueryDataType DOUBLE = new QueryDataType(DoubleConverter.INSTANCE);
+
+    public static final QueryDataType TIME = new QueryDataType(LocalTimeConverter.INSTANCE);
+    public static final QueryDataType DATE = new QueryDataType(LocalDateConverter.INSTANCE);
+    public static final QueryDataType TIMESTAMP = new QueryDataType(LocalDateTimeConverter.INSTANCE);
+    public static final QueryDataType TIMESTAMP_WITH_TZ_DATE = new QueryDataType(DateConverter.INSTANCE);
+    public static final QueryDataType TIMESTAMP_WITH_TZ_CALENDAR = new QueryDataType(CalendarConverter.INSTANCE);
+    public static final QueryDataType TIMESTAMP_WITH_TZ_INSTANT = new QueryDataType(InstantConverter.INSTANCE);
+    public static final QueryDataType TIMESTAMP_WITH_TZ_OFFSET_DATE_TIME = new QueryDataType(OffsetDateTimeConverter.INSTANCE);
+    public static final QueryDataType TIMESTAMP_WITH_TZ_ZONED_DATE_TIME = new QueryDataType(ZonedDateTimeConverter.INSTANCE);
+
+    public static final QueryDataType OBJECT = new QueryDataType(ObjectConverter.INSTANCE);
+
+    /** Converter. */
+    private Converter converter;
+
+    /** Precision. */
+    private int precision;
```

Comment:
If the javadoc is merely duplicating the field name, what is the purpose in having it?

Reply:
Some artifacts from the early prototype. Remove useless JavaDocs here and in other classes.